static/app/views/dashboards/manage/index.tsx

```diff
@@ -63,18 +66,74 @@ type State = {
   dashboards: DashboardListItem[] | null;
   dashboardsPageLinks: string;
   showTemplates: boolean;
+  windowWidth: number;
 } & DeprecatedAsyncView['state'];
 
 class ManageDashboards extends DeprecatedAsyncView<Props, State> {
+  shouldReload = true;
+
+  constructor(props: Props, context: any) {
+    super(props, context);
+
+    this.state = {
+      ...super.getDefaultState(),
+      ...this.getDefaultState(),
+      dashboards: [],
+      dashboardsPageLinks: '',
+      windowWidth: window.innerWidth,
+    };
+  }
+
   getDefaultState() {
     return {
       ...super.getDefaultState(),
       showTemplates: shouldShowTemplates(),
     };
   }
 
+  debouncedHandleResize = debounce(() => {
+    const currentWidth = window.innerWidth;
+    // Only update state if the width has changed
+    if (currentWidth !== this.state.windowWidth) {
+      this.setState({windowWidth: currentWidth});
+    }
+    const paginationObject = parseLinkHeader(this.state.dashboardsPageLinks);
+    if (
+      this.state.dashboards?.length &&
+      paginationObject.next.results &&
+      this.getDashboardsPerPage() > this.state.dashboards.length
+    ) {
+      this.reloadData();
+    }
+  }, 250);
+
+  componentDidMount() {
+    window.addEventListener('resize', this.debouncedHandleResize);
+    super.componentDidMount();
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.debouncedHandleResize);
+    super.componentWillUnmount();
+  }
```

Comment:
I think [`ResizeObserver`](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) would be a better fit here. You could add a `ref` for the element that contains the dashboard grid, and `ResizeObserver` would fire only when that element changes size. _Plus_ it would return the new dimensions of the element. This way you won't have to account for the sidebar width (which changes sometimes!) and you won't need to run `window.innerWidth`, which [causes a window re-paint](https://gist.github.com/paulirish/5d52fb081b3570c81e3a)